packages/core/test/lib/request.test.ts

```diff
@@ -0,0 +1,40 @@
+import { Event } from '@sentry/types';
+
+import { API } from '../../src/api';
+import { eventToSentryRequest } from '../../src/request';
+
+describe('eventToSentryRequest', () => {
+  const api = new API('https://dogsarebadatkeepingsecrets@squirrelchasers.ingest.sentry.io/12312012');
+  const event: Event = {
+    contexts: { trace: { trace_id: '1231201211212012', span_id: '12261980', op: 'pageload' } },
+    environment: 'dogpark',
+    event_id: '0908201304152013',
+    release: 'off.leash.park',
+    spans: [],
+    transaction: '/dogs/are/great/',
+    type: 'transaction',
+    user: { id: '1121', username: 'CharlieDog', ip_address: '11.21.20.12' },
+  };
+
+  it('adds sampling information to transaction item header', () => {
+    event.tags = { samplingMethod: 'client_rate', sampleRate: '0.1121', dog: 'Charlie' };
```

Comment:
I think we should not use tags to implement this.

The tags namespace is controlled by users, we can't just pop off tags like done in this test case.
Just think someone may already have a tag named `"sampleRate"` or may desire to have one in the future, and the SDK shall not mess with that.

Reply:
Yeah, I'm happy to stick it somewhere else. I was just trying to figure out a place to put it that I knew would come through intact, since `eventToSentryRequest` gets the event after all serialization/processing/etc has been done, without adding a lot of machinery.

I can look at it again, and see if there's somewhere else to stick it. Could do a specially-named context, though that is also a spot that users use. Either way, it seems like maybe the real fix here is to use a name users aren't likely to use (`__sentryTracingSampleRate` or something). What do you think?

Reply:
As for the protocol, it's defined in a Notion doc, but didn't cover the cases I listed above. Checking in with @jan-auer about that and will update accordingly.

Reply:
FURTHER UPDATE: Stuff that's happening elsewhere means that I don't think I'll need to piggyback on tags anymore, since the refactoring I was trying to avoid by doing so is now likely going to happen regardless.

Reply:
Can we prefix the tags with `__sentry_xxx` to highlight those are internal?
Then we can also filter out any tag that starts with `__sentry` before sending.

Reply:
>Can we prefix the tags with __sentry_xxx to highlight those are internal?

Yes, though I moved to doing it another way for `tracestate`, and was now planning to just go with a similar solution here.

>Then we can also filter out any tag that starts with __sentry before sending.

Envelopes get their headers after all event processing is done, so this would have to be done in the envelope-header method directly, at which point it doesn't matter where you put it, since you're going to manually delete it anyway.